sentry-android/src/main/java/io/sentry/android/adapters/UnknownAdapter.java

```diff
@@ -0,0 +1,60 @@
+package io.sentry.android.adapters;
+
+import com.google.gson.*;
+import io.sentry.SentryEvent;
+import io.sentry.UnknownField;
+import io.sentry.protocol.SentryId;
+import java.lang.reflect.Type;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class UnknownAdapter<T extends UnknownField> implements JsonDeserializer<T> {
+
+  private Gson gson =
```

Comment:
This partially mirrors the `Gson` instance you register in `AndroidSerializer` and seems to me like a possible cause of troubles (those two instances must be kept in sync)

Reply:
jep, but they cannot have the same adapters nor JsonDeserializationContext or it will be an infinite loop